### PR TITLE
PR for issue #101

### PR DESCRIPTION
**Pull request by peteryxu**
Addressing issue #101

### CODE REVIEW

File: examples/pipeline_human_local_single.py [hunk 1]
```diff
@@ -43,11 +42,13 @@ def get_the_secret_fact() -> str:
...
-            choices=[agent_service.description, human_service.description],
-            components=[agent_component, human_component],
-        )
+        agent_component,
+        human_component
+        # RouterComponent(
+        #     selector=PydanticSingleSelector.from_defaults(llm=OpenAI()),
+        #     choices=[agent_service.description, human_service.description],
+        #     components=[agent_component, human_component],
+        # )
```
Comment (nerdai) [inline examples/pipeline_human_local_single.py:51]: Thanks @peteryxu. Sadly tho, I don't think this fixes the bug. This replaces the desired `RouterComponent` with a deterministic chain of agent_component > human_component. The desired behaviour is instead just one of these components being chosen to handle a task.

I think we might be missing a RouterServiceComponent that is like RouterComponent but takes ServiceComponents as input and is itself a ServiceComponent.

(cc: @logan-markewich)

**Review by nerdai — COMMENTED** [review]
Thanks @peteryxu for the attempt to fix the bug. I think this doesn't quite fix the bug, but replaces it with another (which is valid) pipeline but behaves differently from what we intend still.

File: examples/pipeline_human_local_single.py (hunk 1 above)
Comment (logan-markewich) [reply examples/pipeline_human_local_single.py:51]: I could have sworn this worked. Perhaps I forgot to commit a change or file. Will try and take a look

Comment (peteryxu) [reply examples/pipeline_human_local_single.py:51]: > Thanks @peteryxu. Sadly tho, I don't think this fixes the bug. This replaces the desired `RouterComponent` with a deterministic chain of agent_component > human_component. The desired behaviour is instead just one of these components being chosen to handle a task.
> 
> I think we might be missing a RouterServiceComponent that is like RouterComponent but takes ServiceComponents as input and is itself a ServiceComponent.
> 
> (cc: @logan-markewich)

yeah, QueryPipeline expects ServiceComponent in the chain.. while RouterComponent is NOT a service component

**Comment by peteryxu** [comment]
> Thanks @peteryxu for the attempt to fix the bug. I think this doesn't quite fix the bug, but replaces it with another (which is valid) pipeline but behaves differently from what we intend still.

that makes sense.  btw,   pipeline_local_single.py is NOT using RouterComponent.. 
image

File: examples/pipeline_human_local_single.py (hunk 1 above)
Comment (logan-markewich) [reply examples/pipeline_human_local_single.py:51]: Rip, I think a change jerry made has made this more complicated than it probably needs to be. Trying to smooth this out locally

**Comment by logan-markewich** [comment]
Going to close this in favour of https://github.com/run-llama/llama-agents/pull/118

**Pull request closed without merging**